server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -612,12 +612,46 @@ public static class Builder {
 
             private boolean hasSyntheticSourceCompatibleKeywordField;
 
+            /**
+             * Returns whether the given builder should be used to track source for synthetic source purposes.
+             */
+            private boolean shouldTrackSource(KeywordFieldMapper.Builder keywordBuilder) {
```

Comment:
I am struggling with this name. Isn't this something like `isUsedForSyntheticSourceByParent`?